cloud-media-livestream/keypublisher/README.md

```diff
@@ -0,0 +1,302 @@
+  --display-name="Live Stream Key Publisher API Gateway" \
+  --api=livestream-key-publisher-api \
+  --api-config=livestream-key-publisher-api-config-1 \
+  --location=us-central1
+```
+
+Once created, your API must be enabled. Run a command to describe your API:
+
+```shell
+gcloud api-gateway apis describe livestream-key-publisher-api
+```
+
+You will get the `managedService` field in the response. This is the service you
+need to enable. For example,
+
+```shell
+gcloud services enable livestream-key-publisher-api-144h6p0e7bzc1.apigateway.[MY-PROJECT].cloud.goog
+```
+
+Now you need to locate the endpoint URL of your API. Run a command to describe
+your gateway:
+
+```shell
+gcloud api-gateway gateways describe livestream-key-publisher-api-gateway --location=us-central1
+```
+
+You will get the `defaultHostname` field in the response. For example,
+
+```
+defaultHostname: livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev
+```
+
+You will send API requests to this URL.
+
+Create an API key to use for making queries to your API, and restrict the API
+key such that only your newly created `livestream-key-publisher-api` can be
+called with the API key. Specify the API restriction with the `managedService`
+field from earlier:
+
+```shell
+gcloud alpha services api-keys create --api-target=service=[managedService]
+```
+
+You will get the `keyString` field in the response. This is your API key. Save
+this key, as you will not be able to retrieve it again.
+
+### Test your API
+
+Using the above information, use `curl` to make a `GET` query to your API:
+
+```shell
+curl --location --request POST \
+  'https://livestream-key-publisher-api-gateway-48mwfke9.uc.gateway.dev/keys?api_key=YOUR_API_KEY' \
+  --header 'Content-Type: application/json' \
+  --data '{"mediaId": "my-asset", "provider": "FakeProvider", "keyIds": ["abcd1234", "efgi5678"]}'
+```
+
+Arguments in the request body are:
+
+*   `mediaId`: arbitrary identifier for the media being encrypted.
+*   `provider`: the DRM provider to use. Omit this argument to see a list of
+    supported providers.
+*   `keyIds`: a list of key IDs to prepare for the given media ID.
+
+This example query uses `FakeProvider`. This is a provider used as a reference
+example, which does not make any actual CPIX queries, but generates random hex
+strings in place of a real key. These keys will not work for a real encryption
+setup.
+
+If the query is successful, you will see a response like:
+
+```
+projects/PROJECT_NUMBER/secrets/MEDIA_ID/versions/1
+```
+
+This is the name of the secret version that was created to hold your encryption
+keys.
+
+To verify this, you can use `gcloud` to access the content of that secret:
+
+```shell
+gcloud secrets versions access projects/PROJECT_NUMBER/secrets/MEDIA_ID/versions/1
+```
+
+## Update code
+
+Use `gcloud functions deploy` to update your Cloud Function code. Ensure
+`.env.yaml` still exists and its contents are correct.
```

Comment:
`and its contents are correct.`

Do you expect the user to have somehow deviated/modified the content in the file? If not, leave this part out or simply have the user re-use the example file.

Reply:
sg, I've reference the same example file here now.